### DISCUSSION TITLE
.NET MAUI updates in .NET 9 Preview 4

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#22553] [Announcements]
The team is continuing work on core fundamentals of the .NET MAUI SDK to improve overall product quality. This includes expanding test coverage, end to end scenario testing, and bug fixing. 

Here's a summary of what's new in .NET MAUI in this preview release:

- [New Input Keyboards](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/dotnetmaui.md#new-input-keyboards)
- [Android 15 Beta 1 Support](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/dotnetmaui.md#android-15-beta-1-support)
- [Native AOT for iOS & Mac Catalyst](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/dotnetmaui.md#native-aot-for-ios--mac-catalyst)

.NET MAUI updates in .NET 9 Preview 4:
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/dotnetmaui.md) 
* [GitHub Release](https://github.com/dotnet/maui/releases/tag/9.0.0-preview.4.10690)
* [How to install](https://github.com/dotnet/core/blob/main/release-notes/9.0/install-workloads.md)

.NET 9 Preview 4:
* [Discussion](https://aka.ms/dotnet/9/preview4)
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/README.md)

**Comment by hamiddd1980** [comment]
Thank you for your efforts . but after so many releases of MAUI , I (as a developer from Visual Basic to C# ,WinForm, WPF,ASP.NET,Blazor,...) regret to say that **this is the first product from Microsoft that has a poor support for Right To Left layout in languages** like Arabic ,... and it seems there is no plan to add full support for this capability and requirement:
#14038  : **Graphical objects should be drawn from Right To Left when FlowDirection is RightToLeft**
Thanks again .

**Comment by ziomek64** [comment]
@davidortinau how about fixing #14471 XD
